.nvidia-ci.yml

```diff
@@ -501,114 +502,95 @@ release:ngc-rhel9.6:
+        for file in build-info/${CI_PIPELINE_ID}-${DRIVER_VERSION}-*; do
+          cat "$file" >> build-info/${CI_PIPELINE_ID}-full.txt
+        done
+      done
+      if [ ! -s "build-info/${CI_PIPELINE_ID}-full.txt" ]; then
+        echo "Error: No build information was collected from the release jobs"
+        exit 1
       fi
-      echo "NGCCLI_VERSION ${NGCCLI_VERSION}"
-    - curl -sSLo ngccli_linux.zip https://api.ngc.nvidia.com/v2/resources/nvidia/ngc-apps/ngc_cli/versions/${NGCCLI_VERSION}/files/ngccli_linux.zip
-    - unzip ngccli_linux.zip
-    - chmod u+x ngc-cli/ngc
-
-# .sign forms the base of the deployment jobs which signs images in the CI registry.
-# This is extended with the image name and version to be deployed.
-.sign:ngc:
-  image: ubuntu:latest
-  timeout: 30 minutes
-  when: always
-  stage: sign
+
+.update-nspect:
+  stage: ngc-publish
+  needs:
+    - job: generate-build-info
+      artifacts: true
+  image: 
+    name: "${CNT_NGC_PUBLISH_IMAGE}"
+    pull_policy: always
+  variables:
+    PROJECT_NAME: "gpu-driver-container"
+    REPO_URL: "https://github.com/NVIDIA/gpu-driver-container.git"
+  script:
+    - cnt-ngc-publish nspect --versions-file "build-info/${CI_PIPELINE_ID}-full.txt"
+
+# Update the nspect staging environment to test the nspect publishing logic
+update-nspect-staging:
+  extends:
+    - .update-nspect
+  except:
+    - tags
+  variables:
+    ENV: "stage"
+    RELEASE_VERSION: "test"
+    NSPECT_CLIENT_ID: "${NSPECT_STAGING_CLIENT_ID}"
+    NSPECT_CLIENT_SECRET: "${NSPECT_STAGING_CLIENT_SECRET}"
+
+# Update the nspect production environment with the new release
+update-nspect:
+  extends:
+    - .update-nspect
   rules:
     - if: $CI_COMMIT_TAG
-      variables:
-        DIST: "${SIGN_JOB_NAME}${VERSION}"
   variables:
-    OUT_REGISTRY_USER: "${CI_REGISTRY_USER}"
-    OUT_REGISTRY_TOKEN: "${CI_REGISTRY_PASSWORD}"
-    OUT_REGISTRY: "${CI_REGISTRY}"
-    NGC_CLI_API_KEY: "${NGC_REGISTRY_TOKEN}"
-    IMAGE_NAME: "${NGC_REGISTRY_IMAGE}"
-    IMAGE_TAG: "${CI_COMMIT_TAG}"
-  before_script:
-    - !reference [.ngccli-setup, before_script]
-    - 'if [[ "${PRECOMPILED}" != "true" ]]; then echo DRIVER_VERSION: ${DRIVER_VERSION} ; [[ -n "${DRIVER_VERSION}" ]] || exit 1; fi'
-    - regctl registry login "${OUT_REGISTRY}" -u "${OUT_REGISTRY_USER}" -p "${OUT_REGISTRY_TOKEN}"
-    - |
-      if [[ "${PRECOMPILED}" == "true" ]]; then
-        ./scripts/precompiled.sh version;
-        source kernel_version.txt
-        export IMAGE_TAG=${DRIVER_BRANCH}-${KERNEL_VERSION}-${DIST##*_}
-      else
-          export IMAGE_TAG=${DRIVER_VERSION}-${DIST##*_}
-      fi
-    # We ensure that the IMAGE_NAME and IMAGE_TAG is set
-    - 'echo Image Name: ${IMAGE_NAME} && [[ -n "${IMAGE_NAME}" ]] || exit 1'
-    - 'echo Image Tag: ${IMAGE_TAG} && [[ -n "${IMAGE_TAG}" ]] || exit 1'
+    OSRB_BUG_ID: "${OSRB_BUG_ID}"
+    ENV: "prod"
+    RELEASE_VERSION: "${CI_COMMIT_TAG}"
```

Comment:
@tariq1890 does it make sense to set `RELEASE_VERSION` to `PUBLISH_VERSIONS` instead (which contains the list of driver versions we are publishing)? In the past, we have named the git tags using the date (e.g. DD-MM-YY) so `CI_COMMIT_TAG` may not be the best fit here. The alternative is to change our tagging convention so that our tags contain the driver versions being publish instead of the date of publishing. Thoughts?

Reply:
@JunAr7112 can you see if there is a way to join the `PUBLISH_VERSIONS` string so that is is `_` separated? That way, we can set `RELEASE_VERSION` to this joined string and it will always capture the versions we are publishing.